pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -158,7 +158,7 @@ func makeSegmentstorePodSpec(p *api.PravegaCluster) corev1.PodSpec {
 	return podSpec
 }
 
-func MakeSegmentstoreConfigMap(p *api.PravegaCluster) *corev1.ConfigMap {
+func MakeSegmentstoreConfigMap(p *api.PravegaCluster, cm *corev1.ConfigMap) *corev1.ConfigMap {
```

Comment:
If `SegmentStoreConfigMap` variable is non-nil, it can be directly added as one more element in the corev1.EnvFromSource Array:
https://github.com/pravega/pravega-operator/blob/3250498bd927c7a0df5af52394bcf94309f288d8/pkg/controller/pravega/pravega_segmentstore.go#L70